lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -182,4 +186,62 @@
 
         return -1;
     });
+
+    platform.registerFunction(FunctionsEnum.ArrayFilter, function (callbackfn, thisArg) {
+        // ECMAScript 2017 #sec-array.prototype.filter
+        "use strict";
+
+        if (this === null || this === undefined) {
+            __chakraLibrary.raiseThis_NullOrUndefined("Array.prototype.filter");
+        }
+
+        let o;
+        let len
+        if (__chakraLibrary.isArray(this)) {
+            o = this;
+            len = o.length;
+        } else {
+            o = __chakraLibrary.Object(this);
+            len = __chakraLibrary.GetLength(o);
+        }
+        
+        if (typeof callbackfn != "function") {
+            __chakraLibrary.raiseFunctionArgument_NeedFunction("Array.prototype.filter");
+        }
+
+        let a = __chakraLibrary.arraySpeciesCreate(o, 0);
+        let k = 0;
+        let to = 0;
+
+        if (__chakraLibrary.isArray(a) && thisArg === undefined) {
+            // fast path.
+            while (k < len) {
+                if (k in o) {
+                    let kValue = o[k];
+                    if (callbackfn(kValue, k, o)) {
+                        a[to] = kValue;
```

Comment:
this will still have the problem as Kevin mentioned before - it will call setter when we execute this, no?

Reply:
Yeah, I'm about to update the PR with a version that always calls the __chakraLibrary helper, but I'm running test262 and perf results for it first